modules/manage/pages/kubernetes/k-manage-resources.adoc

```diff
@@ -7,27 +7,55 @@
 
-== Configure CPU resources
+Redpanda is designed to maximize throughput by xref:get-started:architecture.adoc#tpc[pinning its threads to specific CPU cores]. However, CPU pinning is disabled by default in Kubernetes because it uses the https://www.kernel.org/doc/html/latest/scheduler/sched-design-CFS.html[Completely Fair Scheduler (CFS)] to dynamically manage CPU allocation. The CFS distributes CPU time among all processes based on their priority and resource requests, which can lead to frequent context switches and migration of processes across different cores. This behavior can degrade the performance of Redpanda due to:
 
-By default, Redpanda pins its threads to all cores that you allocate to it. The more cores that are allocate to Redpanda, the more throughput it can support. You can configure the CPU resources allocated to each Redpanda broker by overriding the default values. For a description of CPU resource units, see the https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu[Kubernetes documentation^].
+- **Loss of CPU cache affinity**: As processes move across cores, they lose the cache locality, leading to increased cache misses and higher latency.
+- **Scheduling latency**: Dynamic CPU allocation can introduce delays as the scheduler determines which core a process should run on, based on current load and availability.
 
-For production clusters, you should never set `resources.cpu.cores` to anything less than full cores.
+To enable CPU pinning in Kubernetes, you must enable the https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy[`static` policy] for the CPU manager. With this policy enabled, the CPU Manager controls a shared pool of CPUs. Initially, this shared pool contains all the CPUs in the worker node. However, when a container is created with an integral CPU request in a https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed[`Guaranteed` Pod], CPUs for that container are removed from the shared pool and assigned exclusively to that container for its lifetime. Other containers are migrated off these exclusively allocated CPUs.
+
+=== Enable CPU pinning
+
+To enable CPU pinning for Redpanda in Kubernetes:
+
+. Ensure that your worker nodes have enough CPU cores to allocate to Redpanda. Sharing cores with other applications can negate the benefits of CPU pinning.
+
+. Ensure the https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy[`static` policy] is enabled for the CPU Manager on your Kubernetes cluster. With this policy enabled, a request for an integral CPU amount in a `Guaranteed` Pod will result in that container being allocated whole CPUs with a guarantee that no other container will run on them.
++
+[source,bash]
+----
+# Check if the static policy is enabled
+cat /var/lib/kubelet/config.yaml | grep cpuManagerPolicy
```

Comment:
I think we'll want to change this heuristic to something more generally available. Most users won't have access to the kubelet config yaml or won't want to run a privileged pod to get access. It should be sufficient to check if the Pod is in fact running at a Guarenteed QoS and then check if the redpanda Pod can observe whether or not it can see that a cpuset has been assigned to it. It may even be helpful for us to add a check in the container startup so we can log out a _very_ obvious line that says "HEY NO CPUSET FOUND ITS GONNA BE SLOWER".

Reply:
Users need to know that the policy is enabled before they deploy Redpanda. So checking if the Pod is running at a Guarenteed QoS won't work. Should we instead not provide a command and link to the Kubernetes docs for info?

Reply:
That issue is perfect. I wonder if there's a way to check for QoS within a Pod 🤔 

Yeah let's point to k8s docs for now and we can update this once we tackle the linked issue.